From: Ivan Vecera <ivecera@redhat.com>
Date: Thu, 27 Jan 2011 12:57:43 -0500
Subject: [net] be2net: fix missing trans_start update
Message-id: <1296133063-13712-1-git-send-email-ivecera@redhat.com>
Patchwork-id: 32939
O-Subject: [RHEL5.7 PATCH] be2net: fix missing trans_start update
Bugzilla: 671595
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Thomas Graf <tgraf@redhat.com>
RH-Acked-by: Andy Gospodarek <gospo@redhat.com>

BZs:
#671595 - Flapping errors (and panic) with bonding and arp_interval while using be2net included in 2.6.18-238

Description:
This patch fixes problems arising from lack of updates of the trans_start.
This bug arose during the initial backport of the upstream driver. RHEL5
kernel requires the update of trans_start by the driver.

Test status:
https://brewweb.devel.redhat.com/taskinfo?taskID=3061310
Tested by myself and by HP.

Upstream commits:
N/A

Signed-off-by: Ivan Vecera <ivecera@redhat.com>

diff --git a/drivers/net/benet/be_main.c b/drivers/net/benet/be_main.c
index bd99df1..9fb04c5 100644
--- a/drivers/net/benet/be_main.c
+++ b/drivers/net/benet/be_main.c
@@ -534,6 +534,8 @@ static int be_xmit(struct sk_buff *skb,
 
 		be_txq_notify(adapter, txq->id, wrb_cnt);
 
+		netdev->trans_start = jiffies;
+
 		be_tx_stats_update(adapter, wrb_cnt, copied,
 				skb_shinfo(skb)->gso_segs, stopped);
 	} else {
